impeller/renderer/context.h

```diff
@@ -39,6 +39,8 @@ class Context : public std::enable_shared_from_this<Context> {
 
   virtual std::shared_ptr<CommandBuffer> CreateCommandBuffer() const = 0;
 
+  virtual bool WaitUntilCommandsCompleted() = 0;
```

Comment:
I don't think this belongs on `Context` - you want to know when the commands from a particular buffer are completed right?

For example, should this wait for all commands from compute to finish as well?

Reply:
I tried adding a parameter to the SubmitCommands method of CommandBuffer to solve the problem. But I found that in some scenarios, developers can't call SubmitCommands themselves, or even they can't find the Command they want to wait for.

for example
https://github.com/flutter/engine/blob/44649208521a19116330a1d0002d08f3361635a9/shell/common/snapshot_controller_impeller.cc#L66

Reply:
When you call `EntityPass::Render` from `AiksContext::Render` maybe there could be some parameter like a callback for when the texture is done?

Reply:
EntityPass::Render knows the command buffer, and the command buffer has `SubmitCommands(const CompletionCallback& callback)`.

Reply:
@dnfield Thanks you very much! I file a new PR https://github.com/flutter/engine/pull/37772, Any thoughts?